src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs

```diff
@@ -0,0 +1,235 @@
+
+        var instanceWatermarks = arguments.InstanceWatermarks.Select(x => x.Version).ToList();
+
+        IEnumerable<InstanceMetadata> instanceMetadata = await _indexStore.BeginUpdateInstanceAsync(arguments.PartitionKey, instanceWatermarks, CancellationToken.None);
+
+        logger.LogInformation("Completed updating all instance watermarks, Total count {TotalCount}", arguments.InstanceWatermarks.Count);
+
+        return instanceMetadata.Select(x =>
+            new InstanceFileIdentifier
+            {
+                Version = x.VersionedInstanceIdentifier.Version,
+                OriginalVersion = x.InstanceProperties.OriginalVersion,
+                NewVersion = x.InstanceProperties.NewVersion
+            }).ToList();
+    }
+
+    /// <summary>
+    /// Asynchronously batches the instance watermarks and calls the update instance.
+    /// </summary>
+    /// <param name="arguments">BatchUpdateArguments</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>
+    /// A task representing the <see cref="UpdateInstanceBatchAsync"/> operation.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="arguments"/> or <paramref name="logger"/> is <see langword="null"/>.
+    /// </exception>
+    [FunctionName(nameof(UpdateInstanceBatchAsync))]
+    public async Task UpdateInstanceBatchAsync([ActivityTrigger] BatchUpdateArguments arguments, ILogger logger)
+    {
+        EnsureArg.IsNotNull(arguments, nameof(arguments));
+        EnsureArg.IsNotNull(arguments.ChangeDataset, nameof(arguments.ChangeDataset));
+        EnsureArg.IsNotNull(arguments.InstanceWatermarks, nameof(arguments.InstanceWatermarks));
+        EnsureArg.IsNotNull(logger, nameof(logger));
+
+        DicomDataset datasetToUpdate = GetDeserialzedDataset(arguments.ChangeDataset);
+
+        int processed = 0;
+
+        logger.LogInformation("Beginning to update all instance blobs, Total count {TotalCount}", arguments.InstanceWatermarks.Count);
+
+        while (processed < arguments.InstanceWatermarks.Count)
+        {
+            int batchSize = Math.Min(_options.BatchSize, arguments.InstanceWatermarks.Count - processed);
+            var batch = arguments.InstanceWatermarks.Skip(processed).Take(batchSize).ToList();
+
+            if (batch.Count > 0)
+            {
+                logger.LogInformation("Beginning to update instance blobs starting with {StartingRange} and {EndingRange}. Total batchSize {BatchSize}.",
+                    batch[0],
+                    batch[^1],
+                    batchSize);
+
+                await Parallel.ForEachAsync(
+                    batch,
+                    new ParallelOptions
+                    {
+                        CancellationToken = default,
+                        MaxDegreeOfParallelism = _options.MaxParallelThreads,
+                    },
+                    async (instance, token) =>
```

Comment:
nit: you don't need `async` and `await:
```suggestion
                    (instance, token) => _updateInstanceService.UpdateInstanceBlobAsync(instance, datasetToUpdate, token)
```

Reply:
Not sure if that works, I cannot convert a Task to ValueTask